src/Data/Array/Polarized/Push.hs

```diff
@@ -67,6 +74,21 @@ make x n
   | n < 0 = error "Making negative length push array"
   | otherwise = Array (\makeA -> mconcat $ Prelude.replicate n (makeA x))
 
+singleton :: a -> Array a
+singleton x = Array (\writeA -> writeA x)
+
+snoc :: a -> Array a %1-> Array a
+snoc x (Array k) = Array (\writeA -> (k writeA) <> (writeA x))
+
+cons :: a -> Array a %1-> Array a
+cons x (Array k) = Array (\writeA -> (writeA x) <> (k writeA))
+
+foldWith :: Monoid b => (a -> b) -> Array a %1-> b
+foldWith f (Array k) = k f
```

Comment:
This function is `foldMap` (specialized to `Array` and with a linear arrow), so I think we should call it that.

The `Foldable`/`Traversable` issues are likely relevant, but those assume the first parameter to be linear too. Bit unfortunate that we don't have a typeclass for this.